tests/envs/test_envs.py

```diff
@@ -118,21 +114,58 @@ def test_env_determinism_rollout(env_spec: EnvSpec):
     env_2.close()
 
 
-@pytest.mark.parametrize(
-    "spec", gym_testing_env_specs, ids=[spec.id for spec in gym_testing_env_specs]
-)
-def test_render_modes(spec):
-    env = spec.make()
-
-    for mode in env.metadata.get("render_modes", []):
-        if mode != "human":
-            new_env = spec.make(render_mode=mode)
-
-            new_env.reset()
-            new_env.step(new_env.action_space.sample())
-            new_env.render()
-            new_env.close()
-    env.close()
+def check_rendered(rendered_frame, mode: str):
+    """Check that the rendered frame is as expected."""
+    if mode == "rgb_array_list":
+        assert isinstance(rendered_frame, list)
+        for frame in rendered_frame:
+            check_rendered(frame, "rgb_array")
+    elif mode == "rgb_array":
+        assert isinstance(rendered_frame, np.ndarray)
+        assert len(rendered_frame.shape) == 3
+        assert rendered_frame.shape[2] == 3
+        assert np.all(rendered_frame >= 0) and np.all(rendered_frame <= 255)
+    elif mode == "ansi":
+        assert isinstance(rendered_frame, str)
+        assert len(rendered_frame) > 0
+    elif mode == "state_pixels_list":
+        assert isinstance(rendered_frame, list)
+        for frame in rendered_frame:
+            check_rendered(frame, "rgb_array")
+    elif mode == "state_pixels":
+        check_rendered(rendered_frame, "rgb_array")
+    elif mode == "depth_array_list":
+        assert isinstance(rendered_frame, list)
+        for frame in rendered_frame:
+            check_rendered(frame, "depth_array")
+    elif mode == "depth_array":
+        assert isinstance(rendered_frame, np.ndarray)
+        assert len(rendered_frame.shape) == 2
+    else:
+        raise Exception(f"Unknown mode: {mode}")
```

Comment:
Do we want to raise an Exception for unknown modes or just ignore/show a warning?
Even if those are tests for in-gym envs, in the spirit we should just ignore it as long as it is specified in metadata (there is another test for it)

Reply:
I understand what you mean but I think it is more helpful to have errors such that new render modes can be added to test for